### DISCUSSION TITLE
Incompatible version between ChromeDriver and Chrome

### DISCUSSION THREAD

**Discussion by suyangdong** [codemagic-ci-cd/codemagic-docs#1844] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Running a build with `SSH` access on `mac_mini`
2. SSH into the machine and verify the `ChromeDriver` and `Chrome` version

```sh
chromedriver --version
```
> ChromeDriver **111.0.5563.64** (c710e93d5b63b7095afe8c2c17df34408078439d-refs/branch-heads/5563@{#995})

```sh
"/Applications/Google Chrome.app/Contents/MacOS/Google Chrome" --version
```

> Google Chrome **114.0.5735.198**

The `ChromeDriver` and `Chrome` has different version. I ran into the error when I run the Chrome integration test:

```sh
chromedriver --port=4444 &
    flutter config --enable-web
    flutter drive \
    --driver=test_driver/integration_driver.dart \
    --target=integration_test/app_test.dart \
    -d web-server --release --browser-name chrome  
```

> Oops; flutter has exited unexpectedly: "SessionNotCreatedException (33): session
not created: This version of **ChromeDriver** only supports Chrome version **111**
Current browser version is **114.0.5735.198** with binary path /Applications/Google
Chrome.app/Contents/MacOS/Google Chrome
  (Driver info: chromedriver=111.0.5563.64
  (c710e93d5b63b7095afe8c2c17df34408078439d-refs/branch-heads/5563@{#995}),platf
  orm=Mac OS X 13.3.1 arm64)".
A crash report has been written to /Users/builder/clone/flutter_01.log.

### Expected results

I have verified the Chrome and ChromeDriver on my local MacBook Pro which are both **114**, I can run the Chrome integration tests locally. 

CodeMagic's `mac_mini` should also have the **Version 114** of the `ChromeDriver` to successfully run the Chrome integration tests.

### Actual results

CodeMagic `mac_mini`'s `ChromeDriver` version is **111**, while `Chrome`'s version is **114**, and cause the tests to fail. 

### Build id (optional)

64a311f758d2487fd1184fac

**Comment by suyangdong** [comment, marked as answer]
I can get around the problem by running 
```sh
brew upgrade --cask chromedriver
```

It upgrades the `ChromeDriver` to **114**

**Reply by himesh-cm** [reply]
That's great to hear that you were able to resolve it! 
While I’m closing our thread for now, if you have any other questions or suggestions, feel free to reply back or start a new conversation anytime. Have a nice day!

**Reply by suyangdong** [reply]
Thanks Himesh

Can CodeMagic keep the `ChromeDriver` version up to date with the Chrome version, so that users do not have to spend extra dozens of seconds to update it every time?

Thanks!
